autogen/variables.tf

```diff
@@ -106,6 +106,11 @@ variable "network_policy" {
   default     = false
 }
 
+variable "network_policy_provider" {
+  description = "The network policy provider."
+  default     = "PROVIDER_UNSPECIFIED"
```

Comment:
We should probably default to `CALICO` or something which will actually work (unless this automatically works?).

Reply:
`PROVIDER_UNSPECIFIED` is the default value of the underlying resource argument and the tests pass, but we can use `CALICO` if it's a more practical default.

Reply:
Can you do some more digging on what `PROVIDER_UNSPECIFIED` actually does?

If it defaults to the GKE-recommended/supported provider and network policies work, let's keep it.

If it doesn't configure a provider and hence network policies break, let's change to calico.

Reply:
It appears that [Calico is the only supported provider](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy#creating_a_network_policy). I can't find any meaningful documentation on the behaviour of `PROVIDER_UNSPECIFIED`, but converging the stub domains suite with this value yields an enabled network policy with Calico pods on the nodes so it seems to select a default provider.

Reply:
Got it. Let's specify CALICO as the default value then, because it's more explicit and clear.